### ISSUE TITLE
Create a 'related definitions' API

### ISSUE THREAD

**Issue by dabutvin** [clearlydefined/service#402]
I'm thinking similar to how we do 
https://api.clearlydefined.io/suggestions/npm/npmjs/-/glob/7.1.3

We would have another endpoint like 
https://api.clearlydefined.io/related/npm/npmjs/-/glob/7.1.3

And this would have an algorithm that returned a list of coordinates.

We can start by returning other revisions of the same coordinates that have the same data missing

For example:

If `npm/npmjs/-/glob/7.1.3` is missing a declared license, return other `npm/npmjs/-/glob/<revision>` that are also missing a declared license

If `npm/npmjs/-/glob/7.1.3` is missing a source location, return other `npm/npmjs/-/glob/<revision>` that are also missing a source location

```
GET https://api.clearlydefined.io/related/npm/npmjs/-/glob/7.1.3

REPSONSE [ "npm/npmjs/-/glob/7.1.2", "npm/npmjs/-/glob/7.1.0", ... ]
```

**Comment by storrisi** [comment]
I like the approach, but i don't know if returning other revisions of the some component could be considered what we expect as a "relation".

The behavior of adding, or switching, a revision of the same component is already implemented in the definitions list of the UI.

**Comment by dabutvin** [comment]
true. That's a good point. I was thinking of this as sort of the inverse of suggestions.

A suggestion helps you fill in missing data from other versions of a component that have that missing data filled.
A relation helps you find other versions of a component that have similarly missing data for what you are offering.

We can wait and see if anyone else has any input on this, if it is something we actually need/want
